Python/ast.c

```diff
@@ -1850,19 +1837,27 @@ ast_for_comprehension(struct compiling *c, const node *n)
         asdl_seq *t;
         expr_ty expression, first;
         node *for_ch;
+        node *sync_n;
         int is_async = 0;
 
         REQ(n, comp_for);
 
-        if (TYPE(CHILD(n, 0)) == ASYNC) {
+        if (NCH(n) == 2) {
             is_async = 1;
+            REQ(CHILD(n, 0), NAME);
+            assert(strcmp(STR(CHILD(n, 0)), "async") == 0);
```

Comment:
I think we should add another macro:

```
#define REQ_NAME(node, name) REQ(node, NAME); assert(strcmp(node, name) == 0);
```

so that 

```
REQ(CHILD(n, 0), NAME);
assert(strcmp(STR(CHILD(n, 0)), "async") == 0);
```

becomes

```
REQ_NAME(CHILD(n, 0), "async");
```

Reply:
Should that go into Include/node.h? I can add it but I'm not sure it deserves to be in the public API, and it seems out of scope for this PR.

Reply:
IIRC "node.h" isn't part of the public API at all. But it's fine, let's skip this bit.

Reply:
OK, I can make another PR after this one is merged.